app/components/primer/button_component.rb

```diff
@@ -20,28 +20,29 @@ class ButtonComponent < Primer::Component
     DEFAULT_VARIANT = :medium
     VARIANT_MAPPINGS = {
       :small => "btn-sm",
-      DEFAULT_VARIANT => "",
-      :large => "btn-large"
+      DEFAULT_VARIANT => ""
     }.freeze
     VARIANT_OPTIONS = VARIANT_MAPPINGS.keys
 
-    # Icon to be rendered in the button.
+    # Leading visuals appear to the left of the button text and are set to <%= link_to_component(Primer::OcticonComponent) %>.
     #
     # @param system_arguments [Hash] Same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
-    renders_one :icon, lambda { |**system_arguments|
+    renders_one :leading_visual, lambda { |**system_arguments|
       system_arguments[:mr] = 2
 
       Primer::OcticonComponent.new(**system_arguments)
     }
+    alias icon leading_visual # remove alias when all buttons are migrated to new slot names
 
-    # Counter to be rendered in the button.
+    # Trailing visuals appear to the right of the button text and are set to <%= link_to_component(Primer::CounterComponent) %>.
     #
     # @param system_arguments [Hash] Same arguments as <%= link_to_component(Primer::CounterComponent) %>.
-    renders_one :counter, lambda { |**system_arguments|
+    renders_one :trailing_visual, lambda { |**system_arguments|
```

Comment:
What's the reasoning for this rename? Without any context, this API seems less clear. I could see this making sense as a polymorphic slot so we would have `trailing_visual_counter`, for example.

Reply:
the idea is to make it more consistent with other components that also follow the leading/trailing pattern
I wanted to start without the polymorphic at first since we only accept counter, but I guess `trailing_visual_counter` could be ok to do so we can avoid yet another breaking change in the future